frameworks/C++/lithium/lithium.cc

```diff
@@ -54,7 +54,31 @@ void siege(int port) {
 }
 #endif
 
-lru_cache<int, decltype(mmm(s::id = int(), s::randomNumber = int()))> world_cache(10000);
+struct json_cache {
+
+  template <typename T>
+  void insert(T o) { 
+    if (0 == buffer.size())
+      positions.push_back(buffer.size());
+    buffer.append(json_encode(o)); 
+    positions.push_back(buffer.size());
+  }
+  std::string get_json_array(const std::vector<int>& ids) {
+    std::string json = "[";
+    json.append(buffer, positions[ids[0]], positions[ids[0]+1] - positions[ids[0]]);
+    for (int i = 1; i < ids.size(); i++) {
+      json.append(1, ',');
+      json.append(buffer, positions[ids[i]], positions[ids[i]+1] - positions[ids[i]]);
+    }
+    json.append(1, ']');
+    return json;
+  }
+
+  std::string buffer;
+  std::vector<int> positions;
+};
+
+json_cache world_cache;
```

Comment:
It's too early on Monday morning for me to figure out exactly what this cache is doing or why it's needed but if it's caching anything so that subsequent serializations require less work, it's a violation of the last rule in the JSON requirements.

> The serialization to JSON must not be cached

https://github.com/TechEmpower/FrameworkBenchmarks/wiki/Project-Information-Framework-Tests-Overview#json-serialization

Reply:
Sorry @nbrady-techempower  I forgot about this rule. I removed the json caching in my last commit.

Reply:
Thanks! Will merge when it passes travis!